src/core/core.tooltip.js

```diff
@@ -604,11 +608,18 @@ module.exports = function(Chart) {
 			}
 
 			ctx.fillStyle = mergeOpacity(vm.backgroundColor, opacity);
+			ctx.lineWidth = vm.borderWidth;
+			ctx.strokeStyle = vm.borderColor;
+
 			ctx.beginPath();
 			ctx.moveTo(x1, y1);
 			ctx.lineTo(x2, y2);
 			ctx.lineTo(x3, y3);
+			ctx.stroke();
+			ctx.fill();
 			ctx.closePath();
+
+			helpers.drawRoundedRectangle(ctx, ptX, ptY, size.width, size.height, vm.cornerRadius);
```

Comment:
Yeah, it draws the background twice. Maybe it would make sense to just stroke the first rendering of the background but other than that the only solution I could think of was drawing the background and caret together in one path - it would require refactoring both functions and I didn't know if it would be to much alterations on the code. 
One refactor I would like to suggest at least would be calling the drawCaret function inside drawBackground so we don't draw the background inside the drawCaret function...